src/Terrain/MapMaster.java

```diff
@@ -92,15 +99,30 @@ public MapMaster(CommandingOfficers.Commander[] COs, MapInfo mapInfo)
         }
       }
 
-      // Warn if the CO still doesn't have a valid HQ.
-      if( COs[co].HQLocations.isEmpty() )
-      {
-        System.out.println("Warning! CO " + co + " does not have any HQ assigned!");
-        initOK = false;
-        break;
-      }
-    }
-  }
+      if( hasHQ )
+        continue; // If we have an HQ already, we know what our most strategic assets are
+
+      // If we don't have an HQ and do have Labs, then Labs are our HQs
+      if( hasLab )
+        for( XYCoord coord : COs[co].ownedProperties )
+        {
+          if( getEnvironment(coord).terrainType == TerrainType.LAB )
+          {
+            COs[co].HQLocations.add(coord);
+          }
+        }
```

Comment:
So Labs are HQs if there are no HQs? Can one map have both (and are labs useless in that case)?

Reply:
Yep, more or less.

One map can definitely have both:
On AWBW, that results in labs only being useful for lab units. If you define a team that has only labs and no HQs, the game doesn't count them as a viable player.
Here, you can mix teams with only labs or only HQs (which is how the cart works, FWIW).

Lab units/unit bans in general are a future kind of thing, but I do think it can create some neat scenarios.

Reply:
I didn't know the carts even allowed labs with no HQ (save for a couple campaign missions where it could just be a reskin).
I guess that means a team can also have both one or more HQs plus one or more labs, but then the labs don't matter (except for special units). Cool.